sympy/combinatorics/fp_groups.py

```diff
@@ -69,13 +69,59 @@ def __new__(cls, fr_grp, relators):
         obj._is_standardized = False
 
         obj._order = None
+        obj._parent = None
         obj._center = None
         return obj
 
     @property
     def free_group(self):
         return self._free_group
 
+    @property
+    def parent(self):
+        return self._parent
+
+    @property
+    def relators(self):
+        return tuple(self._relators)
+
+    def _generators(self):
+        return self.free_group.generators
+
+    def subgroup(self, gens):
+        '''
+        Return the subgroup generated by `gens` using the
+        Reidemeister-Schreier algorithm
+
+        '''
+        if not all([isinstance(g, FreeGroupElement) for g in gens]):
+            raise ValueError("Generators must be `FreeGroupElement`s")
+        letters = self.free_group.symbols
+        for g in gens:
+            c = [t[0] in letters for t in g.array_form]
+            if not all(c):
```

Comment:
That would be good but that also makes me realise that we can't just make a new `FreeGroup` on already existing `FreeGroupElement`s because then the generators end up with a wrong `.group` attribute. Also, the `group` attribute of the generators returned by `reidemester_presentation()` is a free group on more generators than necessary. Hmm, I'll have another look at it. Having `reidemester_presentation()` return generators with the right `group` attribute and matching relators would probably be better. Alternatively, go through the generators and relators changing their `.group` attribute when a new `FreeGroup` is created on them.